src/rgb/serialize.rs

```diff
@@ -402,6 +402,7 @@ impl csv::serialize::Commitment for rgb::Transition {
     fn commitment_serialize<E: io::Write>(&self, mut e: E) -> Result<usize, csv::serialize::Error> {
         use crate::rgb::commit::Identifiable;
         Ok(
+            self.id.commitment_serialize(&mut e)? +
```

Comment:
Well, this is a commitment serialization; it must have only consensus-critical data. Not sure whether Id is the part of them...

Reply:
it's needed because you don't want to interpret one type of transition as another later. if you don't commit to this value i can change it later

Reply:
Isn't it that the commitment to the serialization order is the same as a commitment to an incremental id?

Reply:
the "id" here has one of the possible types of transitions, like "PRIM_ISSUE_TS" or "TRANSFER_TS". in theory if two transitions have similar fields/seals i could create a transition for one of the two types, but then change it later when i validate it. if you commit to the type this is not possible.

i agree that it could be just the index in a vec, but i prefer to have it explicit with the map just to avoid issues for instance if you add another transition and change the order of the others